k2/csrc/utils.h

```diff
@@ -465,6 +465,14 @@ std::ostream &operator<<(std::ostream &os, const std::vector<T> &vec) {
   return os;
 }
 
+template <typename T>
+struct MaxOp {
+  __host__ __device__ __forceinline__ T operator()(const T &a,
+                                                   const T &b) const {
+    return (a > b) ? a : b;
```

Comment:
Great!  incidentally we may be able to use cub::Max(), not sure if it's the same.

Reply:
`cub_Max` uses `std::numeric_limits<T>::lowest()` as the initial value. Here I use `T(0)` as the initial value (and return T(0) if `num_elems = 0`) according to the documentation you created before.

https://github.com/k2-fsa/k2/blob/e70ce8c7817fd0a8b2842df6d828d2c761e3fd17/k2/csrc/utils.h#L198-L199